packages/core/src/extensions/BubbleMenu/component/BubbleMenu.tsx

```diff
@@ -59,123 +59,157 @@ export const BubbleMenu = (props: { editor: Editor }) => {
     currentBlockListType
   );
 
+  const blockIconMap: Record<string, IconType> = {
+    Text: RiText,
+    "Heading 1": RiH1,
+    "Heading 2": RiH2,
+    "Heading 3": RiH3,
+    "Bullet List": RiListUnordered,
+    "Numbered List": RiListOrdered,
+  };
+
   return (
     <Toolbar>
-      <DropdownMenu trigger={currentBlockName}>
-        <DropdownItemGroup>
-          <DropdownBlockItem
-            title="Text"
-            icon={RiText}
-            isSelected={currentBlockName === "Paragraph"}
-            onClick={() =>
-              props.editor.chain().focus().unsetBlockHeading().unsetList().run()
-            }
-          />
-          <DropdownBlockItem
-            title="Heading 1"
-            icon={RiH1}
-            isSelected={currentBlockName === "Heading 1"}
-            onClick={() =>
+      <SimpleToolbarDropdown
+        text={currentBlockName}
+        icon={blockIconMap[currentBlockName]}
+        items={[
+          {
+            onClick: () => {
+              props.editor.view.focus();
```

Comment:
is this necessary? as there is already a focus in the chain below?

Reply:
For some reason chaining commands still causes that flicker in the bubble menu, even though the editor should be focused before calling the command. It's still definitely a focus related issue, since I had a different fix before using `event.preventDefault()` on `MouseDown` events, but this was a cleaner solution since I also had to deal with some of Mantine Menu's custom behaviour. Either way, I'll remove the `focus()` calls from the chain.